### PR TITLE
chore(ci): add E2E test workflow

### PR DESCRIPTION
**Pull request by odockal**
Adding workflow that runs e2e test suite (#76) after push to merge event and on demand. It does not replace PR check that will be added as well.

### CODE REVIEW

**Comment by odockal** [comment]
@cbr7 If you would please take a look.

**Comment by odockal** [comment]
Fixes #107

File: .github/workflows/e2e-main.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+      - uses: actions/cache@v4
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Execute yarn in SSO Extension
```
Comment (jeffmaury) [inline .github/workflows/e2e-main.yaml:105]:
```suggestion
      - name: Execute yarn in Red Hat Account Extension
```

File: .github/workflows/e2e-main.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        working-directory: ./podman-desktop-redhat-account-ext
+        run: |
+          yarn add -D @podman-desktop/tests-playwright@next
```
Comment (jeffmaury) [inline .github/workflows/e2e-main.yaml:108]: Why is this required ?

File: .github/workflows/e2e-main.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+          yarn --frozen-lockfile
+
+      - name: Run All E2E tests
+        working-directory: ./podman-desktop-redhat-account-ext
+        env:
+          PODMAN_DESKTOP_ARGS: ${{ github.workspace }}/podman-desktop
+        run: yarn test:e2e
+
+      - uses: actions/upload-artifact@v4
+        if: always()
```
Comment (jeffmaury) [inline .github/workflows/e2e-main.yaml:118]: Seems to me that `working-directory: ./podman-desktop-redhat-account-ext` is missing here ?

File: .github/workflows/e2e-main.yaml (hunk 2 above)
Comment (odockal) [reply .github/workflows/e2e-main.yaml:108]: It always pull up the newest version of the dependency. Since it is in lock file, it would have to be update every time manually. This always grabs next tag from npmjs. ATM of "rapid" changes in base tests-playwright  package it seems like good approach.

File: .github/workflows/e2e-main.yaml (hunk 3 above)
Comment (odockal) [reply .github/workflows/e2e-main.yaml:118]: good question. let me check bootc version. given the glob pattern, it might still works.

Comment (odockal) [reply .github/workflows/e2e-main.yaml:118]: works nice in bootc: https://github.com/containers/podman-desktop-extension-bootc/actions/runs/8742851050 even without specifying working dir of the ext.

File: .github/workflows/e2e-main.yaml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  e2e-tests:
+    name: Red Hat Account Extension E2E tests
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        if: github.event_name == 'workflow_dispatch'
+        with:
+          repository: ${{ github.event.inputs.organization }}/${{ github.event.inputs.repositoryName }}
+          ref: ${{ github.event.inputs.branch }}
+          path: ${{ github.event.inputs.repositoryName }}
```
Comment (jeffmaury) [inline .github/workflows/e2e-main.yaml:52]: So the path maybe different (see line 93)

Comment (odockal) [reply .github/workflows/e2e-main.yaml:52]: I see your point now. That needs to be fixed.

Comment (odockal) [reply .github/workflows/e2e-main.yaml:53]: See the step at line 60. I am now using dynamic repository name resolution based on event that triggers the workflow. Variable is used for working-directory now.

**Review by jeffmaury — APPROVED** [review]
LGTM
